Alerting docs: update `Introduction > Notification policies` (#88656)
grafana/grafana: docs/sources/alerting/fundamentals/notifications/notification-policies.md

{{< docs/shared lookup="alerts/how_label_matching_works.md" source="grafana" version="<GRAFANA_VERSION>" >}}

{{< figure src="/media/docs/alerting/notification-routing.png" max-width="750px" caption="Matching alert instances with notification policies" alt="Example of a notification policy tree" >}}

{{% admonition type="note" %}}

To prevent any alerts from being missed, the default notification policy always handles alert instances if there are no child policies or if none of the child policies have any label matchers that match the alert instance's labels.

{{% /admonition %}}

## Routing

To determine which notification policy handles which alert instances, the system start evaluating from the top of the tree, the default notification policy.

If a notification policy has label matchers that match the labels of the alert instance, it descends in to its child policies in the order they are displayed. If there are a matching policy, continues to look for any child policies that might have label matchers that further narrow down the set of labels, and so forth until no more child policies are found.

> As soon as the deepest matching policy is found, the system does not continue to look for sibling policies. If you want to continue to look for other matching policies at the same hierarchical level, enable **Continue matching siblings** on that particular policy.

Reviewer: what is the difference between a sibling and a child - we should write a sentence or two about that

> As soon as the deepest matching policy is found, the system does not continue to look for sibling policies. If you want to continue to look for other matching policies at the same hierarchical level, enable **Continue matching siblings** on that particular policy.

Author: @brendamuir, agree this was also confusing for me  

I am aware about this one- on my TODOs that I commented you in Slack. 

I need to test this to confirm the behaviour and I'll then clarify this a bit more. 

I think the way it works: 
- it starts evaluating siblings (nodes in the same hiercharchical level) in order
  - it says before `it descends in to its child policies in the order they are displayed.`
- when found, it continues evaluating its child nodes
    - it says `if there are a matching policy, continues to look for any child policies`
- when found the deepest node, it stops
  - if `Continue matching siblings` is enabled, evaluates siblings policies (nodes at the same hierarchical level)

> As soon as the deepest matching policy is found, the system does not continue to look for sibling policies. If you want to continue to look for other matching policies at the same hierarchical level, enable **Continue matching siblings** on that particular policy.

Author: I have updated this section again, please, check out.

{{% admonition type="note" %}}
If you haven't configured any label matchers for your notification policy, your notification policy matches _all_ alert instances. This may prevent child policies from being evaluated unless you have enabled **Continue matching siblings** on the notification policy.
{{% /admonition %}}

{{< collapse title="Routing example" >}}

Here's the breakdown of the previous example:

**Pod stuck in CrashLoop** does not have a `severity` label, so none of its child policies are matched. It does have a `team=operations` label, so the first policy is matched.

The `team=security` policy is not a match and **Continue matching siblings** was not configured for that policy.

**Disk Usage – 80%** has both a `team` and `severity` label, and matches a child policy of the operations team.

**Unauthorized log entry** has a `team` label but does not match the first policy (`team=operations`) since the values are not the same, so it will continue searching and match the `team=security` policy. It does not have any child policies, so the additional `severity=high` label is ignored.

{{< /collapse >}}

## Inheritance

> In addition to child policies being a useful concept for routing alert instances, they also inherit properties from their parent policy. This also applies to child policies of the default notification policy.

Reviewer: here we talk about child again...so same thing as above really

Author: https://github.com/grafana/grafana/pull/88656/commits/7d62f0bdabf263a9fc16993e384aec1f9cccffaa tries to clarify about sibling and child policies

    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/fundamentals/notifications/contact-points/
  notification-timings:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/alerting/fundamentals/notifications/group-alert-notifications/#timing-options
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/fundamentals/notifications/group-alert-notifications/#timing-options
  mute-timings:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/alerting/configure-notifications/mute-timings/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/configure-notifications/mute-timings/
  group-alert-notifications:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/alerting/fundamentals/notifications/group-alert-notifications/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/fundamentals/notifications/group-alert-notifications/
---

# Notification policies

> Notification policies provide you with a flexible way to design how to handle notifications and minimize alert noise.

Reviewer: ```suggestion
Notification policies provide you with a flexible way of designing how to handle notifications and minimize alert noise.
```

{{< docs/shared lookup="alerts/how_label_matching_works.md" source="grafana" version="<GRAFANA_VERSION>" >}}

{{< figure src="/media/docs/alerting/notification-routing.png" max-width="750px" caption="Matching alert instances with notification policies" alt="Example of a notification policy tree" >}}

{{% admonition type="note" %}}

To prevent any alerts from being missed, the default notification policy always handles alert instances if there are no child policies or if none of the child policies have any label matchers that match the alert instance's labels.

{{% /admonition %}}

## Routing

To determine which notification policy handles which alert instances, the system starts evaluating from the top of the tree; the default notification policy.

If a notification policy has label matchers that match the labels of the alert instance, it descends in to its child policies in the order they are displayed. If there are a matching policy, continues to look for any child policies that might have label matchers that further narrow down the set of labels, and so forth until no more child policies are found.

> As soon as the deepest matching policy is found, the system does not continue to look for sibling policies. If you want to continue to look for other matching policies at the same hierarchical level, enable **Continue matching siblings** on that particular policy.

Reviewer: ```suggestion
As soon as the deepest matching policy is found, the system does not continue to look for matching policies. If you want to continue to look for other matching policies at the same hierarchical level, enable **Continue matching siblings** on that particular policy.
```

Author: did rewrite this secti9on

**Pod stuck in CrashLoop** does not have a `severity` label, so none of its child policies are matched. It does have a `team=operations` label, so the first policy is matched.

The `team=security` policy is not a match and **Continue matching siblings** was not configured for that policy.

**Disk Usage – 80%** has both a `team` and `severity` label, and matches a child policy of the operations team.

**Unauthorized log entry** has a `team` label but does not match the first policy (`team=operations`) since the values are not the same, so it will continue searching and match the `team=security` policy. It does not have any child policies, so the additional `severity=high` label is ignored.

{{< /collapse >}}

## Inheritance

In addition to child policies being a useful concept for routing alert instances, they also inherit properties from their parent policy. This also applies to child policies of the default notification policy.

By default, a child policy inherits the following notification properties from its parent:

- [Contact point](ref:contact-points)
- [Grouping options](ref:group-alert-notifications)
- [Timing options](ref:notification-timings)

> - [Mute timings](ref:mute-timings)

Reviewer: ```suggestion

```
